java/daikon/chicory/Instrument24.java

```diff
@@ -0,0 +1,1720 @@
+      return null;
+    } else if (binaryClassName.startsWith("jdk.internal.reflect")) {
+      // Starting with Java 9 sun.reflect => jdk.internal.reflect.
+      debug_transform.log(
+          "Ignoring system class %s, in jdk.internal.reflect package", binaryClassName);
+      return null;
+    }
+
+    // Don't instrument our own code.
+    if (isChicory(className)) {
+      debug_transform.log("Not transforming Chicory class %s%n", binaryClassName);
+      return null;
+    }
+
+    ClassLoader cfLoader;
+    if (loader == null) {
+      cfLoader = ClassLoader.getSystemClassLoader();
+      debug_transform.log("Transforming class %s, loader %s - %s%n", className, loader, cfLoader);
+    } else {
+      cfLoader = loader;
+      debug_transform.log(
+          "Transforming class %s, loader %s - %s%n", className, loader, loader.getParent());
+    }
+
+    // Parse the bytes of the classfile, die on any errors.
+    ClassFile classFile =
+        ClassFile.of(
+            ClassFile.ClassHierarchyResolverOption.of(
+                ClassHierarchyResolver.ofResourceParsing(cfLoader)));
+    try {
+      classModel = classFile.parse(classfileBuffer);
+    } catch (Throwable t) {
+      System.err.printf("Unexpected error %s reading in %s%n", t, binaryClassName);
+      t.printStackTrace();
+      // No changes to the bytecodes
+      return null;
+    }
+
+    if (Chicory.dump) {
+      try {
+        debug_transform.log(
+            "Dumping .class and .javap for %s to %s%n", binaryClassName, debug_orig_dir);
+        // Define the output file
+        Path outputFile = Paths.get(debug_orig_dir.toString(), binaryClassName + ".class");
+        // Write the byte array to a .class file.
+        Files.write(outputFile, classFile.transformClass(classModel, ClassTransform.ACCEPT_ALL));
+        // Write a BCEL-like file with an extension of .javap.
+
+        try (BufferedWriter writer =
+            Files.newBufferedWriter(
+                new File(debug_orig_dir, binaryClassName + ".javap").toPath(),
+                StandardCharsets.UTF_8)) {
+          writer.write(classModel.toDebugString());
+        }
+      } catch (Throwable t) {
+        System.err.printf(
+            "Unexpected error %s dumping out debug files for: %s%n", t, binaryClassName);
+        t.printStackTrace();
+        // proceed with instrumentation
+      }
+    }
+
+    hasClinit = false;
+    byte[] newBytes = {};
+    debug_transform.log("%nClass: %s%n", binaryClassName);
+    // Instrument the classfile, die on any errors
+    try {
+      newBytes =
+          classFile.build(
+              classModel.thisClass().asSymbol(),
+              classBuilder -> instrumentClass(classBuilder, classModel, cfLoader));
+
+    } catch (Throwable t) {
+      System.err.printf("Unexpected error %s in transform of %s%n", t, binaryClassName);
+      t.printStackTrace();
+      throw new RuntimeException("Unexpected error", t);
+    }
+
+    if (classInfo.shouldInclude) {
```

Comment:
Can this value be the return value of `instrumentClass` rather than being stored in a field?

Reply:
I don't see how to rewrite the 'newBytes = ..." to do that.